hles/transformation/src/main/scala/org/broadinstitute/monster/dap/RawRecord.scala

```diff
@@ -52,11 +67,13 @@ case class RawRecord(id: Long, fields: Map[String, Array[String]]) {
   /** Get the singleton value for an attribute in this record if one exists, parsed as a boolean. */
   def getOptionalBoolean(field: String): Option[Boolean] = getOptional(field).map(_ == "1")
 
-  /** Get the singleton value for an attribute in this record, parsed as a long. */
+  /** Get the singleton value for an attribute in this record, parsed as a long.
+    * The conversion to double and then long is to handle scientific notation
+    */
   def getOptionalNumber(field: String, truncateDecimals: Boolean = true): Option[Long] =
     getOptional(field).map(value => {
       try {
-        value.toLong
+        value.toDouble.toLong
```

Comment:
:-1: if we convert to a double in this try statement, it'll also handle any decimal values, which will prevent us from logging that we needed to truncate a decimal value and instead simply do so silently.

Reply:
Moved the toDouble into the TruncatedDecimalError logic